auth.js

```diff
@@ -3,732 +3,86 @@
 
 // External modules.
 var _ = require('lodash')
-var Passport = require('passport')
+var AuthUrlmatcher = require('auth-urlmatcher')
+var AuthToken = require('auth-token-cookie')
 
 // Load configuration
 var DefaultOptions = require('./default-options.js')
 
-// External seneca-auth modules
-var AuthToken = require('auth-token-cookie')
-var AuthRedirect = require('auth-redirect')
-var AuthUrlmatcher = require('auth-urlmatcher')
-
 var error = require('eraro')({
   package: 'auth'
 })
 
-module.exports = function auth (options) {
+module.exports = function auth (opts) {
   var seneca = this
 
-  seneca.depends('auth', ['web', 'user'])
+  var internals = {}
+  internals.accepted_framworks = [
+    'express',
+    'hapi'
+  ]
 
   // using seneca.util.deepextend here, as there are sub properties
-  options = seneca.util.deepextend(DefaultOptions, options)
-
-  function migrate_options () {
-    if (options.service || options.sendemail || options.email) {
-      throw error('<' + (options.service ? 'service' : (options.sendemail ? 'sendemail' : 'email')) +
-        '> option is no longer supported, please check seneca-auth documentation for migrating to new version of seneca-auth')
-    }
-
-    if (options.tokenkey) {
-      seneca.log('<tokenkey> option is deprecated, please check seneca-auth documentation for migrating to new version of seneca-auth')
-    }
-  }
-
-  migrate_options()
-  load_default_plugins()
-
-  var m
-  if ((m = options.prefix.match(/^(.*)\/+$/))) {
-    options.prefix = m[1]
-  }
-
-  // define seneca actions
-  // seneca.add({ role:'auth', wrap:'user' },      wrap_user)
-  seneca.add({init: 'auth'}, init)
-
-  seneca.add({role: 'auth', cmd: 'register'}, cmd_register)
-  seneca.add({role: 'auth', cmd: 'user'}, cmd_user)
-  seneca.add({role: 'auth', cmd: 'instance'}, cmd_user)
-  seneca.add({role: 'auth', cmd: 'clean'}, cmd_clean)
-
-  seneca.add({role: 'auth', cmd: 'create_reset'}, cmd_create_reset)
-  seneca.add({role: 'auth', cmd: 'load_reset'}, cmd_load_reset)
-  seneca.add({role: 'auth', cmd: 'execute_reset'}, cmd_execute_reset)
-  seneca.add({role: 'auth', cmd: 'confirm'}, cmd_confirm)
-
-  seneca.add({role: 'auth', cmd: 'update_user'}, cmd_update_user)
-  seneca.add({role: 'auth', cmd: 'change_password'}, cmd_change_password)
+  internals.options = seneca.util.deepextend(DefaultOptions, opts)
 
-  seneca.add({role: 'auth', cmd: 'login'}, cmd_login)
-  seneca.add({role: 'auth', cmd: 'logout'}, cmd_logout)
+  internals.load_default_express_plugins = function () {
+    // External seneca-auth modules
+    var AuthRedirect = require('auth-redirect')
 
-  seneca.add({role: 'auth', cmd: 'register_service'},
-    cmd_register_service)
-
-  seneca.add({role: 'auth', hook: 'map_fields'}, alias_fields)
-
-  function load_default_plugins () {
-    seneca.use(AuthToken)
-    seneca.use(AuthRedirect, options.redirect || {})
+    seneca.use(require('./lib/user-management'), internals.options)
```

Comment:
Please add the require statements to the top of the file... it makes it easier to see what auth.js depends on